src/Esfa.Vacancy.Register.Api/Orchestrators/VacancyOrchestrator.cs

```diff
@@ -1,24 +1,32 @@
-﻿using Esfa.Vacancy.Register.Application.Queries.GetVacancy;
-using MediatR;
-using System.Threading.Tasks;
+﻿using System.Threading.Tasks;
 using AutoMapper;
+using Esfa.Vacancy.Register.Application.Queries.GetVacancy;
+using Esfa.Vacancy.Register.Infrastructure.Settings;
+using MediatR;
 
 namespace Esfa.Vacancy.Register.Api.Orchestrators
 {
-    public class VacancyOrchestrator
+    public class VacancyOrchestrator : IVacancyOrchestrator
     {
         private readonly IMediator _mediator;
+        private readonly string _liveVacancyBaseUrl;
 
-        public VacancyOrchestrator(IMediator mediator)
+        public VacancyOrchestrator(IMediator mediator, IProvideSettings provideSettings)
         {
             _mediator = mediator;
+            _liveVacancyBaseUrl = provideSettings.GetSetting(ApplicationSettingConstants.LiveVacancyBaseUrl);
         }
 
         public async Task<Vacancy.Api.Types.Vacancy> GetVacancyDetailsAsync(int id)
         {
             var response = await _mediator.Send(new GetVacancyRequest() { Reference = id });
+            var vacancy = response == null ? null : Mapper.Map<Vacancy.Api.Types.Vacancy>(response.Vacancy);
 
-            return response == null ? null : Mapper.Map<Vacancy.Api.Types.Vacancy>(response.Vacancy);
+            if (!string.IsNullOrEmpty(_liveVacancyBaseUrl))
```

Comment:
Thinking further about this, could this be part of mapping? Maybe a separate resolver and I don't think we should need to check for the url from settings being null. For the application this must always be set I believe.